fission-cli/library/Fission/CLI/App.hs

```diff
@@ -44,22 +45,33 @@ interpret baseCfg cmd = do
         run' :: FissionCLI errs Connected.Config () -> FissionCLI errs Base.Config ()
         run' = ensureM . Connected.run baseCfg timeoutSeconds
 
-      attempt App.Env.read >>= \case
-        Right Env {appURL} -> do
-          UTF8.putTextLn $ "App already set up at " <> textDisplay appURL
-          logDebug . textDisplay $ AlreadyExists @URL
-          raise $ AlreadyExists @URL
-
-        Left errs -> do
+      attempt CLI.Env.get >>= \case
+        Left errs ->
           case openUnionMatch @(NotFound FilePath) errs of
-            Just _ -> do
-              logDebug @Text "Setting up new app"
-              run' $ Handler.appInit appDir buildDir mayAppName
+            Just err -> do
+              CLI.Error.put err "Fission is installed, but you are not logged in. Please run `fission login`"
+              raise errs
```

Comment:
Hmm we have this exact check in two places just in this PR (maybe more elsewhere?) I imagine that it will be fairly common as we add cases. We can also add some more semantic context to the re-raise (e.g. `ExpectedToBeSetup`, but like... maybe with a better name?)

We have a function elsewhere called `ensureNotLoggedIn` (which makes an HTTP request and whatnot). Perhaps we could extend that convention here. There's a bunch of approaches for this, but a guard may make sense since we're in a context equipped with Rescue:

```haskell
data NotSetup = NotSetup

guardIsSetup :: (MonadRescue m, m`Raises` NotSetup, MonadLogger m) => m ()
guardIsSetup = 
  attempt CLI.Env.get >>= \case
     Left errs ->
       case openUnionMatch @(NotFound FilePath) errs of
          Just err -> do
           CLI.Error.put err "ERROR TEXT HERE"
           raise NotSetup

     Right _ ->
       return ()

foo = do
  actionA
  guardIsSetup
  actionB

-- Or, a bit more flexible *and* handles error in one place --

alreadySetup :: (MonadRescue m, MonadLogger m) => m (Either NotSetup ())
alreadySetup = 
  attempt CLI.Env.get >>= \case
     Left errs ->
       case openUnionMatch @(NotFound FilePath) errs of
          Just err -> return $ Left NotSetup
          Nothing -> return $ Right ()

     Right _ ->
       return $ Right () -- We have an alias `ok = Right ()` that you can use as well

inner = do
  actionX
  ensureM alreadySetup -- ensureM is from Rescue
  actionY

outer = do
  actionA
  actionB
  attempt inner >>= \case -- Handle all of them in one place
    Right _ ->
      actionC
      actionD

    Left err ->
      case openUnionMatch @NotSetup errs of
        Just NotSetup -> CLI.Error.put err "ERROR TEXT HERE"
        Nothing -> return ()
```

The second version essentially pushes the checking and messaging to the caller (in our case, I believe that's `Fission.CLI.App.interpret`).